test/parallel/test-stream-preprocess.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const fs = require('fs');
+const path = require('path');
+const rl = require('readline');
+
+const BOM = '\uFEFF';
+
+// get the data by a non-stream way to compare with the streamed data
+const modelData = fs.readFileSync(
+  path.join(__dirname, '..', 'fixtures', 'file-to-read-without-bom.txt'), 'utf8'
+);
+const modelDataFirstCharacter = modelData[0];
+
+// detect the number of forthcoming 'line' events for mustCall() 'expected' arg
+const linesNumber = modelData.match(/\n/g).length;
+
+// ensure both without-bom and with-bom test files are textwise equal
+assert.strictEqual(
+  fs.readFileSync(
+    path.join(__dirname, '..', 'fixtures', 'file-to-read-with-bom.txt'), 'utf8'
+  ),
+  `${BOM}${modelData}`
+);
+
+// an unjustified BOM stripping with a non-BOM character unshifted to a stream
+const inputWithoutBOM = fs.createReadStream(
+  path.join(__dirname, '..', 'fixtures', 'file-to-read-without-bom.txt'), 'utf8'
+);
+
+inputWithoutBOM.once('readable', common.mustCall(() => {
+  const maybeBOM = inputWithoutBOM.read(1);
+  assert.strictEqual(maybeBOM, modelDataFirstCharacter);
+
+  if (maybeBOM !== BOM) inputWithoutBOM.unshift(maybeBOM);
```

Comment:
In this test script, it should be pretty much known whether `maybeBOM !== BOM` is true, so I think you can turn these into assertions.